addons/web/static/src/core/datetime/datetime_picker.scss

```
@@ -21,80 +21,83 @@
             var(--DateTimePicker__Cell-border-radius-end)
             var(--DateTimePicker__Cell-border-radius-end)
             var(--DateTimePicker__Cell-border-radius-start);
+        position: relative;
     }
 
-    .o_highlighted {
-        --bg-opacity: 0.25;
+    .o_datetime_button span {
+        z-index: 2;
     }
 
     .o_selected {
-        --bg-opacity: 0.5;
+        &,.o_current {
+            color: $o-component-active-color;
+        }
+        
+        &:not(.o_select_start, .o_select_end) {
+            background: $o-component-active-bg;
+        }
     }
 
-    .o_current {
-        --bg-opacity: 0.75;
+    .o_current:not(.o_select_start,.o_select_end),
+    .o_highlight_start:not(.o_selected),
+    .o_highlight_end:not(.o_selected),
+    .o_select_start, .o_select_end {
```

Since you're using one line for each selector, I suggest
 
```suggestion
    .o_select_start,
    .o_select_end {
```